### PR TITLE
python-modernize . -w -f libmodernize.fixes.fix_dict_six

### PR DESCRIPTION
**Pull request by nickpell**
@dimagi/py3

### CODE REVIEW

File: corehq/apps/app_manager/fixtures/mobile_ucr.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def uses_reports(self, restore_state):
             return False
 
         apps = self._get_apps(restore_state, restore_user)
-        report_configs = self._get_report_configs(apps).values()
+        report_configs = list(self._get_report_configs(apps).values())
         if not report_configs:
             return False
```
Comment (millerdev) [inline corehq/apps/app_manager/fixtures/mobile_ucr.py]: Probably not something you want to fix now, but this could be simplified to
```py
return bool(self._get_report_configs(apps))
```
 The `.values()` call is not necessary for how the result is being used.

File: corehq/apps/app_manager/models.py [hunk 2]
```diff
@@ -415,10 +415,10 @@ class FormActions(DocumentSchema):
 
     def all_property_names(self):
         names = set()
-        names.update(self.update_case.update.keys())
-        names.update(self.case_preload.preload.values())
+        names.update(list(self.update_case.update.keys()))
```
Comment (millerdev) [inline corehq/apps/app_manager/models.py]: I think most of these can be changed to eliminate `.keys()` (and sometimes even `list(...)`), so this would become
```py
names.update(self.update_case.update)
```
In general `list(dict_instance)` is equivalent to `list(dict_instance.keys())`

Understood if this is not something you want to address now since these are automated changes.

File: corehq/apps/app_manager/suite_xml/generator.py [hunk 3]
```diff
@@ -111,7 +111,7 @@ def media_resources(self):
             for lang in self.build_profile.langs:
                 media_list += self.app.media_language_map[lang].media_refs
             requested_media = set(media_list)
-        for path, m in sorted(self.app.multimedia_map.items(), key=lambda item: item[0]):
+        for path, m in sorted(list(self.app.multimedia_map.items()), key=lambda item: item[0]):
```
Comment (millerdev) [inline corehq/apps/app_manager/suite_xml/generator.py]: unnecessary change

File: corehq/apps/app_manager/suite_xml/post_process/instances.py [hunk 4]
```diff
@@ -123,7 +123,7 @@ def __call__(self, fn):
 }
 
 
-@register_factory(*INSTANCE_KWARGS_BY_ID.keys())
+@register_factory(*list(INSTANCE_KWARGS_BY_ID.keys()))
```
Comment (millerdev) [inline corehq/apps/app_manager/suite_xml/post_process/instances.py]: how does this even work (even before this change)? the args would be passed in a non-deterministic order. Is there only ever one arg?

**Review by millerdev — APPROVED** [review]
Didn't review every line. I think fine to merge if you don't want to visit and clean up each change manually.

Edit: however, it might be nice to eliminate the obviously unnecessary changes.

File: corehq/apps/app_manager/suite_xml/post_process/instances.py (hunk 4 above)
Comment (nickpell) [reply corehq/apps/app_manager/suite_xml/post_process/instances.py]: it looks like the ```__init__``` for ```register_factory``` could take in the args in any order and ```__call__``` would have the same result.

File: corehq/apps/app_manager/fixtures/mobile_ucr.py (hunk 1 above)
Comment (nickpell) [reply corehq/apps/app_manager/fixtures/mobile_ucr.py]: https://github.com/dimagi/commcare-hq/pull/19033

**Comment by nickpell** [comment]
@millerdev I'm going to leave most of the automated changes as-is for now.  I may return to do some clean up if I'm bored later.
